packages/sdk/src/solana/programs/nft-drop.ts

```diff
@@ -166,34 +170,45 @@ export class NFTDrop {
    * console.log(nfts[0].metadata.name);
    * ```
    */
-  async getAll(): Promise<NFT[]> {
+  async getAll(queryParams?: QueryAllParams): Promise<NFT[]> {
+    const parsedQueryParams = QueryAllParamsSchema.parse(queryParams);
+
     // TODO: Add pagination to get NFT functions
     const info = await this.getCandyMachine();
-    const claimed = await this.getAllClaimed();
+    const claimed = await this.getAllClaimed(parsedQueryParams);
```

Comment:
this wont work i think :( coz you're paginating the get all, then paginating the lazy minted items and "hoping" that the 2 slices will match but there's no guarantee... esspecially since claiming mints a random lazyminted item in the list

Reply:
Hmm I have an idea

Reply:
Replaced it with a new method which should solve this - first you get all the claimed NFTs you can and fill the beginning of count w/ it, then any remaining count gets filled with unclaimed